lib/json_schema/validator.rb

```diff
@@ -511,9 +501,8 @@ def validate_strict_properties(schema, data, errors, path)
     end
 
     def validate_type(schema, data, errors, path)
-      return true if !schema.type || schema.type.empty?
-      valid_types = schema.type.flat_map { |t| TYPE_MAP[t] }.compact
-      if valid_types.any? { |t| data.is_a?(t) }
+      return true if schema.type.empty?
```

Comment:
Any reason to remove the check on `!schema.type`? It might still be wise to hedge against a `nil` here.